pkg/mount/mountutil.go

```diff
@@ -112,6 +115,77 @@ func getDiskDevPath(partition string) (string, error) {
 	return "/dev/" + parentDisk, nil
 }
 
+//	getSoftLinkForPartition returns path to /sys/class/block/$partition
+//	if the path does not exist and the partition is "root"
+//	then the root partition is detected from /proc/cmdline
+func getSoftLinkForPartition(partition string) (string, error) {
+	softlink := getLinkForPartition(partition)
+
+	if _, err := os.Stat(softlink); os.IsNotExist(err) && partition == "root" {
```

Comment:
question: Are there cases where `/dev/root` is symlink to the root partition?

Reply:
```shell
$ readlink -f /dev/root
/dev/root
```

it is not like a regular symlink. if the /dev/root is a real device, then /sys/class/block/root will exist and the original code will execute. the fallback checks if /sys/class/block/root does not exists.

Reply:
I was referring to this answer. Though I am also not sure, whether it will be a symlink in any kernel version.